src/disp/program.c

```diff
@@ -815,6 +822,72 @@ static MVMuint32 value_index_how(MVMThreadContext *tc, MVMDispProgramRecording *
     return MVM_VECTOR_ELEMS(rec->values) - 1;
 }
 
+/* Ensures we have a values used entry for the specified unboxed int. */
+static MVMuint32 value_index_unbox_int(MVMThreadContext *tc, MVMDispProgramRecording *rec,
+        MVMuint32 from_value) {
+    /* Look for an existing such value. */
+    MVMuint32 i;
+    for (i = 0; i < MVM_VECTOR_ELEMS(rec->values); i++) {
+        MVMDispProgramRecordingValue *v = &(rec->values[i]);
+        if (v->source == MVMDispProgramRecordingUnboxValue &&
+                v->unbox.from_value == from_value)
```

Comment:
I think this should check `v->unbox.kind == MVM_CALLSITE_ARG_INT` too; we do have allomorphs and a dispatch program that unboxes the same thing two ways is possible.